core/db/entities.py

```diff
@@ -109,6 +111,41 @@ class Project(db.Model):
     package = sqla.orm.relationship("Package", back_populates="projects")
 
 
+class ProjectDeliveryPlan(db.Model):
+    """Stores Project Delivery Plan data for projects."""
+
+    __tablename__ = "project_delivery_plan"
+
+    id = sqla.Column(GUID(), default=uuid.uuid4, primary_key=True)  # this should be UUIDType once using Postgres
+    milestone = sqla.Column(sqla.String(), nullable=False)
+    project_id = sqla.Column(sqla.String(), sqla.ForeignKey("project_dim.id"), nullable=False)
+    start_date = sqla.Column(sqla.DateTime(), nullable=False)
+    end_date = sqla.Column(sqla.DateTime(), nullable=False)
+    status = sqla.Column(sqla.Enum(StatusEnum, name="project_delivery_plan_status"), nullable=False)
+
+    # TODO: Should this be nullable? Is Status alone enough in some circumstances?
+    comments = sqla.Column(sqla.String(), nullable=False)
```

Comment:
we should probably add a unique constraint to prevent duplicate data being entered. I would suggest an index composed of milestone and project_id. Then we can comment with the intention that data model people can check it?

Reply:
yeah good spot I missed that!